securesystemslib/metadata.py

```diff
@@ -97,3 +98,85 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:  # pragma: no cover
+        """sign the payload and create the signature
+
+        Arguments:
+            signer: A "Signer" class instance
+
+        Raises:
+            TypeError: If "signer" is not a instance of "Signer" class.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        if not isinstance(signer, Signer):
+            raise TypeError(f"expected type 'Signer', got {type(signer).__name__}")
+
+        signature = signer.sign(self.pae)
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, key: Key) -> bool:  # pragma: no cover
+        """verify the payload with provided verifier
+
+        Arguments:
+            key: A "Key" class instance
+
+        Raises:
+            TypeError: If "key" is not a instance of "Key" class.
+
+        Returns:
+            # To be confirmed.
```

Comment:
Current wrappers returns either boolean value or throughs error in verification but DSSE protocol states is should parse the `payload` according to given `payloadType`. So, does that means secure systems lib have to maintain a list of supported payload types.
in-toto ITE-5 states it will have a `payloadType` as `application/vnd.in-toto+json`.

Reply:
> I don't exactly know how those two are related?

I just want to confirm return type as it is not specified in [DSSE Protocol](https://github.com/secure-systems-lab/dsse/blob/master/protocol.md#protocol) verify method. Does DSSE verification have to return serialized body or none.

Suggest some exceptions to throw in verification, and maybe I have to create new exceptions then suggest some names for them.

Reply:
I think I got the right return type, it must be `recognised_signers` which is a list of names of verified signers.